From 79e4a6272058a0ab640b4d7617054c541e8b6191 Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Tue, 28 Feb 2012 12:06:10 +0530
Subject: [PATCH 131/182] ARM: OMAP2+: mcasp: add missing HWMOD data

HWMOD data regarding edma interrupts, sysconfig for McASP were
missing. This patch adds the same.

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c |   27 ++++++++++++++++++++++++++-
 1 files changed, 26 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index 31ab0fb..4fe0196 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -1415,8 +1415,18 @@ static struct omap_hwmod am33xx_mailbox_hwmod = {
 };
 
 /* 'mcasp' class */
+
+static struct omap_hwmod_class_sysconfig am33xx_mcasp_sysc = {
+	.rev_offs	= 0x0,
+	.sysc_offs	= 0x4,
+	.sysc_flags	= SYSC_HAS_SIDLEMODE,
+	.idlemodes	= (SIDLE_FORCE | SIDLE_NO | SIDLE_SMART),
+	.sysc_fields	= &omap_hwmod_sysc_type3,
+};
+
 static struct omap_hwmod_class am33xx_mcasp_hwmod_class = {
-	.name		= "mcasp",
+	.name	= "mcasp",
+	.sysc	= &am33xx_mcasp_sysc,
 };
 
 /* mcasp0 */
@@ -1426,6 +1436,12 @@ static struct omap_hwmod_irq_info am33xx_mcasp0_irqs[] = {
 	{ .irq = -1 }
 };
 
+static struct omap_hwmod_dma_info am33xx_mcasp0_edma_reqs[] = {
+	{ .name = "tx", .dma_req = AM33XX_DMA_MCASP0_X, },
+	{ .name = "rx", .dma_req = AM33XX_DMA_MCASP0_R, },
+	{ .dma_req = -1 }
+};
+
 static struct omap_hwmod_addr_space am33xx_mcasp0_addr_space[] = {
 	{
 		.pa_start	= 0x48038000,
@@ -1452,6 +1468,7 @@ static struct omap_hwmod am33xx_mcasp0_hwmod = {
 	.class		= &am33xx_mcasp_hwmod_class,
 	.clkdm_name	= "l3s_clkdm",
 	.mpu_irqs	= am33xx_mcasp0_irqs,
+	.sdma_reqs	= am33xx_mcasp0_edma_reqs,
 	.main_clk	= "mcasp0_fck",
 	.prcm		= {
 		.omap4	= {
@@ -1470,6 +1487,13 @@ static struct omap_hwmod_irq_info am33xx_mcasp1_irqs[] = {
 	{ .irq = -1 }
 };
 
+static struct omap_hwmod_dma_info am33xx_mcasp1_edma_reqs[] = {
+	{ .name = "tx", .dma_req = AM33XX_DMA_MCASP1_X, },
+	{ .name = "rx", .dma_req = AM33XX_DMA_MCASP1_R, },
+	{ .dma_req = -1 }
+};
+
+
 static struct omap_hwmod_addr_space am33xx_mcasp1_addr_space[] = {
 	{
 		.pa_start	= 0x4803C000,
@@ -1496,6 +1520,7 @@ static struct omap_hwmod am33xx_mcasp1_hwmod = {
 	.class		= &am33xx_mcasp_hwmod_class,
 	.clkdm_name	= "l3s_clkdm",
 	.mpu_irqs	= am33xx_mcasp1_irqs,
+	.sdma_reqs	= am33xx_mcasp1_edma_reqs,
 	.main_clk	= "mcasp1_fck",
 	.prcm		= {
 		.omap4	= {
-- 
1.7.9.1

